### PR TITLE
fix wechatgame short black screen after first screen

### PR DESCRIPTION
**Pull request by novain1102**
Re: #https://github.com/cocos/cocos-engine/issues/15013

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: templates/wechatgame/first-screen.js [hunk 1]
```diff
@@ -370,6 +380,16 @@ function draw() {
     drawTexture(gl, program, sloganTexture, sloganVertexBuffer, 4);
     // draw progress bar
     drawProgressBar(gl, programProgress, vertexBufferProgress, 3, progress, progressBarColor, progressBackground);
+
+    if (enableScissorTest) {
```
Comment (novain1102) [inline templates/wechatgame/first-screen.js:387]: Restore the state before modification after rendering

File: templates/wechatgame/game.ejs [hunk 2]
```diff
@@ -24,7 +25,7 @@ if (canvas){
     canvas.height = _h;
 }
 // Adjust initial canvas size
-if (canvas && window.devicePixelRatio >= 2) {canvas.width *= 2; canvas.height *= 2;}
+if (canvas && window.devicePixelRatio >= 2) {canvas.width *= info.devicePixelRatio; canvas.height *= info.devicePixelRatio;}
```
Comment (novain1102) [inline templates/wechatgame/game.ejs:28]: Fix possible dislocation of the first screen rendering

File: templates/wechatgame/game.ejs [hunk 3]
```diff
@@ -64,7 +65,12 @@ function onApplicationCreated(application) {
         require('./engine-adapter');
         return application.init(cc);
     }).then(() => {
-        return firstScreen.end().then(() => application.start());
+        return firstScreen.setProgress(0.8).then(() => {
+            cc.director.once(cc.Director.EVENT_BEGIN_FRAME, () => {
```
Comment (novain1102) [inline templates/wechatgame/game.ejs:69]: Call first.end() to stop firstScreen rendering until the engine renders to avoid black screen

**Review by SantyWang — APPROVED** [review]

**Review by minggo — APPROVED** [review]

File: templates/wechatgame/first-screen.js [hunk 4]
```diff
@@ -177,8 +180,8 @@ function initProgressVertexBuffer() {
 }
 
 function updateVertexBuffer() {
-    const widthRatio = image.width / canvas.width;
-    const heightRatio = image.height / canvas.height;
+    const widthRatio = image.width / canvas.width * 1.35;
+    const heightRatio = image.height / canvas.height * 1.35;
```
Comment (minggo) [inline templates/wechatgame/first-screen.js:186]: Why have the magic number?

**Comment by minggo** [comment]
The codes use many magic number. Please add comment for them or use a meaningful variable name.

File: templates/wechatgame/first-screen.js (hunk 4 above)
Comment (novain1102) [reply templates/wechatgame/first-screen.js:186]: firstScreen is rendered according to the pixels of the provided image, and the size parameters of the display are adjusted according to the provided drawing by UI designer.

Comment (minggo) [reply templates/wechatgame/first-screen.js:186]: Yep, it is better to add the information in codes. Or we will forget why need do it like this.

Comment (novain1102) [reply templates/wechatgame/first-screen.js:186]: got it.

**Comment by novain1102** [comment]
> The codes use many magic number. Please add comment for them or use a meaningful variable name.

Add ratio value comment.

**Comment by zetaztt** [comment]
``` typescript
cc.director.once(cc.Director.EVENT_BEGIN_FRAME, () => {
  firstScreen.end();
});
application.start();
```
这段代码在安卓小程序上没有问题，但是在IOS上会报错，无法进入游戏，如果我直接调用就是正常的
``` typescript
firstScreen.end();
application.start();
```

**Pull request merged**